generators/readme/test/generate-readme.spec.ts

```diff
@@ -91,3 +96,117 @@ describe('generate README', () => {
+      path.join(testPackages, 'justice-league')
+    );
+    const content = file.toString();
+    expect(content).toMatchInlineSnapshot(`
+      "<!-- THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. -->
+      <!-- This file is created by the \`yarn generate-readme\` script. -->
+
+      # JusticeLeague
+
+      ## Description
+
+      Render an Justice League
+
+      ## Installation
+
+      \`\`\`
+      yarn add @commercetools-test/justice-league
+      \`\`\`
+
+      \`\`\`
+      npm --save install @commercetools-test/justice-league
+      \`\`\`
+
+      Additionally install the peer dependencies (if not present)
+
+      \`\`\`
+      yarn add react
+      \`\`\`
+
+      \`\`\`
+      npm --save install react
+      \`\`\`
+
+      ## Usage
+
+      \`\`\`jsx
+      /* eslint disable */
+      import React from 'react';
+      import JusticeLeague from '@commercetools-uikit/justice-league';
+
+      const Example = () => <JusticeLeague name=\\"Batman\\" />;
+
+      export default Example;
+
+      \`\`\`
+
+      ## Properties
+
+      | Props | Type | Required | Default | Description |
+      | - | - | :-: | - | - |
+      | \`name\` | \`string\` | ✅ | | The name of an Avenger. |
+      | \`identity\` | \`Object\`<br/>[See signature.](#signature-identity) | ✅ | | The real identity of this Avenger, if known. |
+      | \`onClick\` | \`(event: Event) => void\` | | | A callback function, called when the component is clicked. |
+      | \`abilities\` | \`Array: Ability[]\`<br/>[See signature.](#signature-abilities) | ✅ | | List the abilities of this Avenger. |
+      | \`movies\` | \`union\`<br/>Possible values:<br/>\`string , Movie\`<br/>[See signature.](#signature-movies) | ✅ | | The list of movies where this Avenger appears in.&#xA;It can either be just the name of the movie or a more detailed&#xA;information about the movie. |
+      | \`power\` | \`union\`<br/>Possible values:<br/>\`1 , 2 , 3 , 4 , 5 , 6 , 7 , 8 , 9 , 10\` | | \`1\` | Define the power for this Avenger, from a scale of 1-10. |
+      | \`isAlive\` | \`boolean\` | | \`true\` | |
+
+      ## Signatures
+
+      ### Signature identity
+
+      \`\`\`ts
+      {
+        /**
+         * The name of the Avenger.
+         */
+        firstName?: string;
+        lastName?: string;
+        age?: number;
+      }
+      \`\`\`
+
+      ### Signature abilities
+
+      \`\`\`ts
+      {
+        /**
+         * The name of the ability.
+         */
+        name: string;
+        description?: string;
+        /**
+         * Set this to \`true\` to mark this ability as new.
+         *
+         * @@defaultValue@@: false
+         */
+        isNew?: boolean;
+      }
+      \`\`\`
+
+      ### Signature movies
+
+      \`\`\`ts
+      {
+        /**
+         * The title of the movie.
+         */
+        title: string;
+        /**
+         * The release date of the movie (ISO).
+         */
+        releaseDate: string;
+      }
+      \`\`\`
```

Comment:
Note that we "embed" the signature types like this, and link to them from the props table.
This way we keep the table simple, while having the type signatures properly documented.